### PR TITLE
Support migrated apps when migration is disabled

### LINKED ISSUE #146: add suffix/prefix to name of kapp created metadata configmaps to avoid configmap naming collisions
Add the ability to add a default prefix for all kapp application's state `configMap` to differentiate them from other `configMap` stored in the same namespace so it won't conflict with the application's `configMap` :smile_cat: 

- [x]  add suffix apps.k14s.io to config map
- [x]  add functionality for users to opt in to this feature
- [x]  manage backward compatibility 
- [x]  if an app is deployed with this feature opted in via the FQ name env variable, the app should continue to behave like the features has been opted in even if the env flag is unset for subsequent kapp actions on the same app.

---
_(edit: hoist solution for those who searched for the corresponding error message)_
👉🏻  TL;DR. https://carvel.dev/kapp/docs/v0.53.0/configmap-migration/

### PR DESCRIPTION
**Pull request by praveenrewar**
- Apps migrated to use the ConfigMap suffix should be supported (seen without suffix) even if migration is disabled
- Remove `isKappApp` check as we don't have that with existing rename and update as well. If we want to, we can add it in all places later on.

`Expected behaviour`
```bash
$ KAPP_FQ_CONFIGMAP_NAMES=true kapp deploy -a my-app -f app.yml --yes


$ KAPP_FQ_CONFIGMAP_NAMES=true kapp ls
Namespace  Name        Namespaces  Lcs   Lca
default    my-app      default     true  7s

$ kapp ls
Namespace  Name        Namespaces  Lcs   Lca
default    my-app      default     true  7s

$ kapp inspect -a my-app
...
Resources in app 'my-app'

Namespace  Name  Kind       Owner  Rs  Ri  Age
...
```

`Note`: This leads to an additional call to get configmap with fully qualified name when migration is disabled.

### CODE REVIEW

File: pkg/kapp/app/recorded_app.go [hunk 1]
```diff
@@ -133,38 +133,36 @@ func (a *RecordedApp) CreateOrUpdate(labels map[string]string) error {
...
-		}
+	configMapWithSuffix, err := a.coreClient.CoreV1().ConfigMaps(a.nsName).Get(context.TODO(), a.fqName, metav1.GetOptions{})
+	if err == nil {
+		return a.updateApp(configMapWithSuffix, labels)
+	}
 
+	if errors.IsNotFound(err) {
 		configmapWithoutSuffix, err := a.coreClient.CoreV1().ConfigMaps(a.nsName).Get(context.TODO(), a.name, metav1.GetOptions{})
 		if err == nil {
-			if a.isKappApp(configmapWithoutSuffix) {
```
Comment (neil-hickey) [inline pkg/kapp/app/recorded_app.go:151]: Do we want to keep the isKappApp check? It looks like we dropped it

File: pkg/kapp/app/recorded_app.go [hunk 2]
```diff
@@ -457,11 +435,6 @@ func (a *RecordedApp) labeledApp() (*LabeledApp, error) {
 	return &LabeledApp{sel, a.identifiedResources}, nil
 }
 
-func (a *RecordedApp) isKappApp(app *corev1.ConfigMap) bool {
```
Comment (neil-hickey) [inline pkg/kapp/app/recorded_app.go:460]: Same as above -  just wondering if the assumptions changed here - or why we dropped this

Comment (neil-hickey) [reply pkg/kapp/app/recorded_app.go:460]: Actually I think I answered my own question - we are assuming the suffix has meaning and is a kapp app by nature right?

**Review by neil-hickey — APPROVED** [review]
LGTM 👍

**Review by sethiyash — APPROVED** [review]
Looks Good.

**Review by kumaritanushree — APPROVED** [review]

File: pkg/kapp/app/recorded_app.go (hunk 2 above)
Comment (praveenrewar) [reply pkg/kapp/app/recorded_app.go:460]: Yep, we are assuming that the suffix is internal and people will not use it.
Also, another reason to remove the `isKappApp` check was to make it consistent with current deploy and rename.
> Remove isKappApp check as we don't have that with existing rename and update as well. If we want to, we can add it in all places later on.

**Pull request merged**